extract.py

```diff
@@ -120,6 +148,81 @@ def extract_database(args):
     return output_rows
 
 
+def extract_csv(args, conf):
+    output_rows = []
+    report = get_report()
+    person_id = conf.get("initial_id", 0)
+    with open(args.source, "r") as datasource:
+        rows = csv.DictReader(datasource)
+        for row in rows:
+            handled_row = translate_row(row, report, conf)
+            if not handled_row[0]:
+                handled_row[0] = person_id
+                person_id += 1
+            output_rows.append(handled_row)
+
+    shuffle(output_rows)
+    if args.verbose:
+        print_report(report)
+    return output_rows
+
+
+def translate_row(row, report, conf):
+    output_row = []
+    value_maps = conf["value_mapping_rules"]
+    column_maps = conf["translation_map"]
+
+    record_id = translation_lookup(row, "record_id", column_maps)
+    output_row.append(record_id)
+
+    given_name = translation_lookup(row, "given_name", column_maps)
+    validate(report, "given_name", given_name, value_maps)
+    clean_given_name = clean_string(given_name)
+    output_row.append(
+        value_maps.get("given_name", {}).get(clean_given_name, clean_given_name)
+    )
+
+    family_name = translation_lookup(row, "family_name", column_maps)
+    validate(report, "family_name", family_name, value_maps)
+    clean_family_name = clean_string(family_name)
+    output_row.append(
+        value_maps.get("family_name", {}).get(clean_family_name, clean_family_name)
+    )
+
+    dob = translation_lookup(row, "DOB", column_maps)
+    dob = clean_dob_fromstr(dob, conf["date_format"])
+    validate(report, "DOB", dob, value_maps)
+    output_row.append(value_maps.get("DOB", {}).get(dob, dob))
+
+    sex = translation_lookup(row, "sex", column_maps)
+    validate(report, "sex", sex, value_maps)
+    output_row.append(value_maps.get("sex", {}).get(sex, sex))
+
+    # is phone or phone_number the canonical field name?
+    phone_number = translation_lookup(row, "phone", column_maps)
+    validate(report, "phone", phone_number, value_maps)
+    clean_phone_number = clean_phone(phone_number)
+    output_row.append(
+        value_maps.get("phone", {}).get(clean_phone_number, clean_phone_number)
+    )
+
+    household_street_address = translation_lookup(row, "address", column_maps)
+    validate(report, "address", household_street_address, value_maps)
+    clean_household_street_address = clean_string(household_street_address)
+    output_row.append(
+        value_maps.get("address", {}).get(
+            clean_household_street_address, clean_household_street_address
+        )
+    )
+
+    household_zip = translation_lookup(row, "zip", column_maps)
+    validate(report, "household_zip", household_zip, value_maps)
```

Comment:
```suggestion
    household_zip = translation_lookup(row, "zip", column_maps)
    validate(report, "zip", household_zip, value_maps)
```

As I noted on some of the other validate calls, this should be zip.

Otherwise if I included a mapping like:


```json
"value_mapping_rules": {
    ....
    "zip": {"A":"B"}
    }
```

Unregularizable/unmappable fields aren't detected and reported.

I'm a little concerned that this bug occurs silently and has popped up a few times. I'd love to see `validate` do some initial checking to ensure that the field is valid or a unit test. I'm ok saving additional testing for a future PR for the sake of expediency.

Reply:
I'm also not really a fan of using the variable `household_zip` instead of `zip`. I think that's part of the confusion here, although I recognize that name was already used in the `HEADER` constant prior to this PR.

Reply:
I'm not sure if this is a bug in the way you've categorized it-more a confusion of namespaces. The header (which governs the values in the report) uses one set of terms. The set of names used by the script internally are given by [OLD_DATA_DICTIONARY](https://github.com/mitre/data-owner-tools/blob/csv-translation/utils/data_reader.py#L11) in `utils/data_reader.py` and are slightly different. I agree we should probably align them, but that's a outside the scope of this feature IMO

Reply:
The `field` argument isn't just used as a header in the report. It's also used to look up regularization rules.

https://github.com/mitre/data-owner-tools/blob/31f01f60b27949e472dc5153fc64de2e04101ad1/extract.py#L85-L86 It's also used 

The bug is that we're trying to find regularization rules under the wrong key in the configuration file.